pytorch_lightning/trainer/trainer.py

```diff
@@ -767,6 +796,45 @@ def run_test(self):
 
         return eval_loop_results
 
+    def run_predict(self):
+        # prepare dataloaders
+        dataloaders, max_batches = self.predict_loop.get_predict_dataloaders(None)
+
+        # check if we want to skip this evaluation
+        if self.predict_loop.should_skip_predict(dataloaders, max_batches):
+            return []
+
+        # ref model
+        model = self.get_model()
+
+        # enable eval mode + no grads
+        self.predict_loop.on_predict_model_eval()
+        model.zero_grad()
```

Comment:
Do we need this? Is this in case the user somehow is doing optimizer.step() in their code?

```suggestion
        model.zero_grad()
```

Reply:
In predict mode ? I think it should be fine.